test/syscalls/linux/uname.cc

```diff
@@ -42,7 +42,7 @@ TEST(UnameTest, SetNames) {
   SKIP_IF(!ASSERT_NO_ERRNO_AND_VALUE(HaveCapability(CAP_SYS_ADMIN)));
 
   constexpr char kHostname[] = "wubbalubba";
-  ASSERT_THAT(sethostname(kHostname, sizeof(kHostname)), SyscallSucceeds());
+  ASSERT_THAT(sethostname(kHostname, sizeof(kHostname)-1), SyscallSucceeds());
```

Comment:
I don't think that we need to change existing test cases. It is better to add a new one. In addition, I would suggest to check these cases:
sethostname("0123456789\0", 3);
sethostname("0123456789\0xxx", 11);
sethostname("0123456789\0xxx", 12);

Reply:
Thanks for the advice. Have changed the test cases. One more question, what's the difference for last two cases:
sethostname("0123456789\0xxx", 11);
sethostname("0123456789\0xxx", 12);

Reply:
> difference for last two cases:
> sethostname("0123456789\0xxx", 11);
sethostname("0123456789\0", 11);

The buffer contains the terminating null character and it is the last one.

> sethostname("0123456789\0xxx", 12);
sethostname("0123456789\0xxx", 12)

The buffer contains the terminating null character and it isn't the last one. gethostname() should return "0123456789" for this case.

Reply:
Got that, they are indeed different test cases.